### ISSUE TITLE
Auto Completion for keywords and extra_keywords in other languages, such as python.

### ISSUE THREAD

**Issue by equiclouds** [microsoft/monaco-editor#833]
When language=html, the editor can suggetion words of all html tags, such as html, section, div, and so on.
But when language=python, the editor can't suggention keywords, such as def, from, import, list, dict, etc.
And We may custom other keywords.

Can options paramter of editor.create function define a key: **extra_keywords**, to make custom global suggentions?

`{ extra_keywords: "Server Request Response Session" }`

This make it easy to edit some code with specified frameworks, such as Flask, OpenCV, IronPython, etc..

**Comment by alexandrudima** [comment]
Here is an example on how to provide custom completion items:

https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-completion-provider-example

**Comment by bnuzhouwei** [comment]
The example above define a key express, when we typed express, it will auto type "express", but when we type exp, the word express does not show in hint list.

I want we define a keywords list, when I type the first few letters, I can pick up the words in suggestion list, that's IntelliSense.

Is that possible. 

Best regards!

**Comment by alexandrudima** [comment]
Yes, that is possible, the sample shows how to create a completion item provider.

All the suggestions ever rendered in the monaco editor (including the ones you mention from HTML) use that exact same API.

**Comment by sally2015** [comment]
@rcjsuen How to apply  `python-language-server` to your simple demo，thanks

**Comment by arogozhnikov** [comment]
@sally2015 
I was stuck with this for a while, but the only thing I needed to do is 

1. install pyls (python-language-server)
2. change [this line](
https://github.com/TypeFox/monaco-languageclient/blob/master/example/src/json-server-launcher.ts#L19) in language-client example to 
```
const serverConnection = server.createServerProcess('JSON', 'pyls');
```

and then this example worked for me out-of-the-box.

**Comment by arogozhnikov** [comment]
@anshullodha 
delete this json and start typing some python - you'll see it making python suggestions and code analysis.

**Comment by anshullodha** [comment]
@arogozhnikov I tried, its not working.

**Comment by cbchien** [comment]
@anshullodha I am not able to reproduce a working monaco with pyls. Can you see if my steps are what you did?

1. install pyls (python-language-server) using pip
2. git clone https://github.com/TypeFox/monaco-languageclient.git 
3. `npm install` under `monaco-languageclient`
4. change the following line in monaco-languageclient `example/src/json-server-launcher.ts` to
```
const serverConnection = server.createServerProcess('JSON', 'pyls');
```
5. npm install
6. npm run build
7. npm run start:ext

I am getting messages like this
```
JSON Server: usage: pyls [-h] [--tcp] [--host HOST] [--port PORT]
            [--log-config LOG_CONFIG | --log-file LOG_FILE] [-v]
pyls: error: unrecognized arguments: C:\Users\user\Desktop\monaco-languageclient\example\lib\ext-json-server.js
```

Python 3.6.4

Any suggestion or notes on what may have been missed?

**Comment by rcjsuen** [comment]
@cbchien Please share your complete `example/src/json-server-launcher.ts` file.

**Comment by cbchien** [comment]
@rcjsuen I had `const serverConnection = server.createServerProcess('JSON', 'pyls');`

**Comment by wlnirvana** [comment]
Any update on this thread? I think I should comment here instead of opening a new issue, but please let me know if I should have done that.

I believe the OP wanted a completion provider that is able to suggest

1. words that have been typed in by a user -  this is the default behavior.
2. some built-in keywords - like `Server Request Response Session`

However, following the [example][1] given by @rcjsuen, I was able to get `Server Request Response Session` suggestions, but the built-in word based suggestion is completely gone, as illustrated in this [stackoverflow question][2].

[1]:https://github.com/Microsoft/monaco-editor/issues/833#issuecomment-383541656
[2]:https://stackoverflow.com/q/51359689

Is there any way to work around this?

**Comment by wufeng87** [comment]
@wlnirvana  I got the same question too.

**Comment by wufeng87** [comment]
@wlnirvana https://github.com/Microsoft/monaco-editor/issues/1036

**Comment by maranranu** [comment]
@arogozhnikov I tried your solution to add python suppot in monaco editor.
Changed 

> const serverConnection = server.createServerProcess('JSON', 'node ', [extJsonServerPath]);  
to 

> const serverConnection = server.createServerProcess('JSON', 'pyls');
After npm install , npm run build && npm run start:ext, I am getting error on console 
sample Language Client: [Error - 15:26:59] Server initialization failed.
main.bundle.js:formatted:10977 Sample Language Client: Error: Connection got disposed.

I debug a lot still no luck.
